android/PartialSync/app/src/main/java/io/realm/todo/ItemsActivity.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.todo;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.helper.ItemTouchHelper;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.EditText;
+
+import io.realm.Realm;
+import io.realm.Sort;
+import io.realm.SyncUser;
+import io.realm.todo.model.Item;
+import io.realm.todo.model.Project;
+import io.realm.todo.ui.ItemsRecyclerAdapter;
+
+public class ItemsActivity extends AppCompatActivity {
+
+    private Realm mRealm;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_items);
+
+        setSupportActionBar(findViewById(R.id.toolbar));
+
+        String projectId = getIntent().getStringExtra("project_id");
+
+        findViewById(R.id.fab).setOnClickListener(view -> {
+            View dialogView = LayoutInflater.from(this).inflate(R.layout.dialog_task, null);
+            EditText taskText = dialogView.findViewById(R.id.task);
+            new AlertDialog.Builder(ItemsActivity.this)
+                    .setTitle("Add a new task")
+                    .setMessage("What do you want to do next?")
+                    .setView(dialogView)
+                    .setPositiveButton("Add", (dialog, which) -> mRealm.executeTransactionAsync(realm -> {
+                        Item item = new Item();
+                        item.setBody(taskText.getText().toString());
+                        realm.where(Project.class).equalTo("id", projectId).findFirst().getTasks().add(item);
+                    }))
+                    .setNegativeButton("Cancel", null)
+                    .create()
+                    .show();
+        });
+
+        mRealm = Realm.getDefaultInstance();
+        Project project = mRealm.where(Project.class).equalTo("id", projectId).findFirst();
```

Comment:
Is it intentional that you want this query and the recycler one to be synchronous? Asynchronous behavior would be "safer", but also result in a more complicated flow.

Reply:
It's a trade-off, I want to keep the flow simple, otherwise, you can't sort directly on the RealmList like 
```Java
project.getTasks().sort("timestamp", Sort.ASCENDING)
```
since the `getTasks` is not ready yet when using async